app/src/main/java/com/example/mpweather/MainActivity.kt

```diff
@@ -28,21 +29,18 @@ class MainActivity : AppCompatActivity() {
     private val appid = "62985350695c6c5b6f476b823778d5e5"
 
     // Display tempiture in two decimal format design
-    var df = DecimalFormat("#.##")
+    internal var decimalFormatting = DecimalFormat("#.##")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        //Get handles for views
-        etCity = findViewById(R.id.etCity)
-        etCountry = findViewById(R.id.etCountry)
-        tvResult = findViewById(R.id.tvResult)
+
     }
 
     fun getWeatherDetails(view: View?) {
         //Define a string to hold the complete url
         var tempUrl = ""
-        val city = etCity!!.text.toString().trim { it <= ' ' }
-        val country = etCountry!!.text.toString().trim { it <= ' ' }
+        val city = etCity.text.toString()
+        val country = etCountry.text.toString()
```

Comment:
This change is not necessary, you can use either of them, I did this because I did not understand the functionality of trim.